File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        l,r = 0,len(numbers)-1
        while l < r:
            curSum = numbers[l] + numbers[r]
            if curSum == target:
                return [l+1,r+1]
            elif curSum > target:
                r -= 1
            else:
                l += 1
            
        
        

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l, r;
        l = 0, r = numbers.size() - 1;
        vector<int> res;

        while (l < r) {
            if (numbers[l] + numbers[r] == target) {
                res.push_back(l + 1);
                res.push_back(r + 1);
                break;
            }
            else if (numbers[l] + numbers[r] < target) l++;
            else r--;
        }
        return res;
    }
};